File: src/core/models/scheduler_events.rs
use crate::core::models::{BreakType, SchedulerStatus};

use chrono::{DateTime, Utc};

#[derive(Debug, Clone)]
pub struct BreakStartedEventArgs {
    pub break_type: BreakType,
    pub duration_seconds: i32,
    pub activity_index: i32,
}

#[derive(Debug, Clone)]
pub struct BreakEndedEventArgs {
    pub break_type: BreakType,
    pub completed: bool,
    pub skipped: bool,
    pub snoozed: bool,
}

#[derive(Debug, Clone)]
pub struct SchedulerSnapshot {
    pub status: SchedulerStatus,
    pub is_paused_manual: bool,
    pub eye_remaining: chrono::Duration,
    pub move_remaining: chrono::Duration,
    pub eye_enabled: bool,
    pub move_enabled: bool,
    pub active_break: Option<BreakType>,
    pub queued_break: Option<BreakType>,
    pub active_break_remaining: Option<chrono::Duration>,
    pub next_break_due: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone)]
pub struct SchedulerDiagnostics {
    pub status: SchedulerStatus,
    pub eye_remaining_seconds: i64,
    pub move_remaining_seconds: i64,
    pub eye_next_due: DateTime<Utc>,
    pub move_next_due: DateTime<Utc>,
    pub eye_snooze_until: Option<DateTime<Utc>>,
    pub move_snooze_until: Option<DateTime<Utc>>,
    pub is_paused_manual: bool,
    pub is_idle_paused: bool,
    pub is_configuration_paused: bool,
    pub is_outside_work_hours: bool,
    pub active_break: Option<BreakType>,
    pub queued_break: Option<BreakType>,
    pub glass_clarity: i32,
    pub background_tint_opacity: f64,
    pub blur_radius: f64,
    pub idle_seconds: f64,
    pub snapshot_captured: bool,
    pub eye_completed: i32,
    pub move_completed: i32,
    pub skipped: i32,
    pub snoozed: i32,
}

File: .برنامه ویندوزی/AWAYRA-WPF-main (2)/src/Awayra.Core/Models/SchedulerEvents.cs
namespace Awayra.Core.Models;

public sealed class SchedulerSnapshot
{
    public SchedulerStatus Status { get; init; }
    public bool IsPausedManual { get; init; }
    public TimeSpan EyeRemaining { get; init; }
    public TimeSpan MoveRemaining { get; init; }
    public bool EyeEnabled { get; init; }
    public bool MoveEnabled { get; init; }
    public BreakType? ActiveBreak { get; init; }
    public BreakType? QueuedBreak { get; init; }
    public TimeSpan? ActiveBreakRemaining { get; init; }
    public DateTimeOffset? NextBreakDue { get; init; }
}

public sealed class BreakStartedEventArgs : EventArgs
{
    public required BreakType BreakType { get; init; }
    public required int DurationSeconds { get; init; }
    public required int ActivityIndex { get; init; }
}

public sealed class BreakEndedEventArgs : EventArgs
{
    public required BreakType BreakType { get; init; }
    public required bool Completed { get; init; }
    public required bool Skipped { get; init; }
    public required bool Snoozed { get; init; }
}
